File: app/src/main/java/com/codingblocks/restapi/adapters/AlbumsAdapter.java
package com.codingblocks.restapi.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.codingblocks.restapi.R;
import com.codingblocks.restapi.models.Album;

import java.util.ArrayList;

/**
 * Created by championswimmer on 12/07/17.
 */

public class AlbumsAdapter extends RecyclerView.Adapter<AlbumsAdapter.AlbumViewHolder> {

    private ArrayList<Album> albums;

    public AlbumsAdapter(ArrayList<Album> albums) {
        this.albums = albums;
    }

    public void setAlbums(ArrayList<Album> albums) {
        this.albums = albums;
        notifyDataSetChanged();
    }

    @Override
    public AlbumViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater li = LayoutInflater.from(parent.getContext());
        return new AlbumViewHolder(li.inflate(R.layout.list_item_album, parent, false));
    }

    @Override
    public void onBindViewHolder(AlbumViewHolder holder, int position) {
        holder.bindView(albums.get(position));
    }

    @Override
    public int getItemCount() {
        return albums.size();
    }

    class AlbumViewHolder extends RecyclerView.ViewHolder {
        TextView tvAlbumTitle;

        public AlbumViewHolder(View itemView) {
            super(itemView);
            tvAlbumTitle = (TextView) itemView.findViewById(R.id.tvAlbumTitle);
        }

        void bindView(Album album) {
            tvAlbumTitle.setText(album.getTitle());
        }
    }
}

File: app/src/main/java/com/codingblocks/restapi/adapters/AlbumsAdapter.kt
package com.codingblocks.restapi.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.codingblocks.restapi.R
import com.codingblocks.restapi.models.Album

import kotlinx.android.synthetic.main.list_item_album.view.*

import java.util.ArrayList

/**
 * Created by championswimmer on 12/07/17.
 */

class AlbumsAdapter(private var albums: ArrayList<Album>?) : RecyclerView.Adapter<AlbumsAdapter.AlbumViewHolder>() {

    fun setAlbums(albums: ArrayList<Album>) {
        this.albums = albums
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AlbumViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_album, parent, false)

        return AlbumViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: AlbumViewHolder, position: Int) {
        holder.bindView(albums!![position])
    }

    override fun getItemCount(): Int {
        return (albums?.size ?: 0)
    }

    inner class AlbumViewHolder(val albumView: View) : RecyclerView.ViewHolder(albumView) {


        fun bindView(album: Album) {
            albumView.tvAlbumTitle.text = album.title
        }
    }
}
